Added better handling for empty tables

miversen33 opened the pull request:
Addresses #146 

Note, I don't really know how to write a test to properly address the change in logic applied here. However I did run `make` and all tests passed.

That said, its worth considering if this is how you want to address this issue or not :)

olimorris commented:
This is a great spot and exactly what is happening. Thank you.

Can you add your test from #146 as well?

miversen33 commented:
> This is a great spot and exactly what is happening. Thank you.
> 
> Can you add your test from #146 as well?

Technically the test I provide there fails as the logic in that test is _ever so slightly_ different from what I am actually doing. That said, I will see if I toss something akin to it in here for you :)

miversen33 commented:
@olimorris I have added a modified version of the test I recommended earlier.

That said, I really don't like it. The logic added is actually in `init` not `dirs_match`, but there is no test (from what I saw) for `init.allowed_dir` so what I added is my best shot at it.

The test is not really clear on what its doing so I added some comments. Tests shouldn't need comments though :(

It works but I would advise you peek at it and make it how you want
